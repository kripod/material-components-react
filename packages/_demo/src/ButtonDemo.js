import Button from '@react-material/button';
import React from 'react';

const ButtonDemo = () =>
  <div>
    <h1>Button</h1>

    {Object.entries({
      light: '',
      dark: 'mdc-theme--dark'
    }).map(([themeName, themeStyleClassName], i) =>
      <div key={i}>
        <h2>{themeName} theme</h2>
        <section className={themeStyleClassName}>
          {[true, false].map((isEnabled, j) =>
            <fieldset key={j} disabled={!isEnabled}>
              <legend>{isEnabled ? 'Enabled': 'Disabled'}</legend>

              <Button>Default</Button>
              <Button raised>Raised</Button>
              <Button dense>Dense Default</Button>
              <Button dense raised>Dense Raised</Button>
              <Button compact>Compact</Button>
              <Button compact raised>Compact Raised</Button>
              <Button primary>Default with Primary</Button>
              <Button raised primary>Raised with Primary</Button>
              <Button accent>Default with Accent</Button>
              <Button raised accent>Raised with Accent</Button>
            </fieldset>
          )}
        </section>
      </div>
    )}
  </div>
;

export default ButtonDemo;
