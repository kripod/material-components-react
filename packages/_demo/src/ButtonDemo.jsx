import React from 'react';
import Button from '../../button';

const ButtonDemo = () =>
  <div>
    <h1>Button</h1>

    <section>
      {[true, false].map(isEnabled =>
        <fieldset key={isEnabled} disabled={!isEnabled}>
          <legend>{isEnabled ? 'Enabled' : 'Disabled'}</legend>

          <Button>Default</Button>
          <Button raised>Raised</Button>
          <Button dense>Dense Default</Button>
          <Button dense raised>Dense Raised</Button>
          <Button compact>Compact</Button>
          <Button compact raised>Compact Raised</Button>
          <Button primary>Default with Primary</Button>
          <Button raised primary>Raised with Primary</Button>
          <Button accent>Default with Accent</Button>
          <Button raised accent>Raised with Accent</Button>
        </fieldset>,
      )}
    </section>
  </div>
;

export default ButtonDemo;
