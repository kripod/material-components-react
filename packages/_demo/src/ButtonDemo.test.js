import React from 'react';
import renderer from 'react-test-renderer';
import ButtonDemo from './ButtonDemo';

it('renders correctly', () => {
  const tree = renderer.create(<ButtonDemo />).toJSON();
  expect(tree).toMatchSnapshot();
});
