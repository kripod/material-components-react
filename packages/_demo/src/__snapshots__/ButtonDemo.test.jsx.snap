exports[`test renders correctly 1`] = `
<div>
  <h1>
    Button
  </h1>
  <section>
    <fieldset
      disabled={false}>
      <legend>
        Enabled
      </legend>
      <button
        className="mdc-button ">
        Default
      </button>
      <button
        className="mdc-button mdc-button--raised ">
        Raised
      </button>
      <button
        className="mdc-button mdc-button--dense ">
        Dense Default
      </button>
      <button
        className="mdc-button mdc-button--dense mdc-button--raised ">
        Dense Raised
      </button>
      <button
        className="mdc-button mdc-button--compact ">
        Compact
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--compact ">
        Compact Raised
      </button>
      <button
        className="mdc-button mdc-button--primary ">
        Default with Primary
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--primary ">
        Raised with Primary
      </button>
      <button
        className="mdc-button mdc-button--accent ">
        Default with Accent
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--accent ">
        Raised with Accent
      </button>
    </fieldset>
    <fieldset
      disabled={true}>
      <legend>
        Disabled
      </legend>
      <button
        className="mdc-button ">
        Default
      </button>
      <button
        className="mdc-button mdc-button--raised ">
        Raised
      </button>
      <button
        className="mdc-button mdc-button--dense ">
        Dense Default
      </button>
      <button
        className="mdc-button mdc-button--dense mdc-button--raised ">
        Dense Raised
      </button>
      <button
        className="mdc-button mdc-button--compact ">
        Compact
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--compact ">
        Compact Raised
      </button>
      <button
        className="mdc-button mdc-button--primary ">
        Default with Primary
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--primary ">
        Raised with Primary
      </button>
      <button
        className="mdc-button mdc-button--accent ">
        Default with Accent
      </button>
      <button
        className="mdc-button mdc-button--raised mdc-button--accent ">
        Raised with Accent
      </button>
    </fieldset>
  </section>
</div>
`;
